plugin/pkg/scheduler/algorithm/priorities/priorities.go

```diff
@@ -28,38 +29,69 @@ import (
 
 // the unused capacity is calculated on a scale of 0-10
 // 0 being the lowest priority and 10 being the highest
-func calculateScore(requested, capacity int64, node string) int {
+func calculateScore(requested int64, capacity int64, node string) int {
 	if capacity == 0 {
 		return 0
 	}
 	if requested > capacity {
-		glog.Infof("Combined requested resources from existing pods exceeds capacity on minion: %s", node)
+		glog.Infof("Combined requested resources %d from existing pods exceeds capacity %d on minion: %s",
+			requested, capacity, node)
 		return 0
 	}
 	return int(((capacity - requested) * 10) / capacity)
 }
 
-// Calculate the occupancy on a node.  'node' has information about the resources on the node.
+// For each of these resources, a pod that doesn't request the resource explicitly
+// will be treated as having requested the amount indicated below, for the purpose
+// of computing priority only. This ensures that when scheduling zero-limit pods, such
+// pods will not all be scheduled to the machine with the smallest in-use limit,
+// and that when scheduling regular pods, such pods will not see zero-limit pods as
+// consuming no resources whatsoever.
+const defaultMilliCpuLimit int64 = 100  // 0.1 core
+const defaultMemoryLimit int64 = 60 * 1024 * 1024  // 60 MB
+
+// TODO: Consider setting default as a fixed fraction of machine capacity (take "capacity api.ResourceList"
+// as an additional argument here) rather than using constants
+func toNonzeroLimits(limits *api.ResourceList) (int64, int64) {
```

Comment:
getLimits() hides the fact that it adjusts the limits, so I prefer getNonZeroLimits(). Changed it to that.

I generally favor passing in the most narrow type possible, in this case api.ResourceList rather than the entire container.  As for what to return, resourceRequest is probably better but that is defined in the predicates package and as it stands now this code doesn't depend on predicates. So I'd want to move that somewhere else (like maybe pkg/api/resource_helpers.go)... but I think all of this code needs to be rewritten so I'm not sure it's worth doing a lot of work.